src/auth/s2s/serviceAuth.ts

```diff
@@ -0,0 +1,19 @@
+import { totp } from 'node-otp'
+import { http } from '../../http/http'
+import { S2SConfig } from './s2sConfig.interface'
+
+// TODO: To be replaced with a proper logging library
+const logger = console
+
+export async function postS2SLease(s2sConfig: S2SConfig): Promise<string> {
+    const oneTimePassword = totp({ secret: s2sConfig.s2sSecret })
+
+    logger.info('generating from secret: ', s2sConfig.s2sSecret, s2sConfig.microservice, oneTimePassword)
```

Comment:
Don't think logging out secrets and passwords is a good idea for code to be checked in ;)

Reply:
Agreed - now removed :)